activerecord/lib/active_record/persistence.rb

```diff
@@ -525,6 +529,12 @@ def touch(*names, time: nil)
       end
     end
 
+  protected
+
+    def actually_destroyed?
+      @_actually_destroyed ||= false
```

Comment:
It might make sense to clarify `@destroyed` to e.g. `@surreptitiously_destroyed` so it doesn't fight for the same meaning as `@_actually_destroyed`.